code/core/src/main/java/com/adobe/marketing/mobile/internal/DataMarshaller.kt

```diff
@@ -62,26 +63,57 @@ internal object DataMarshaller {
      * @param intent [Intent]
      * @param marshalledData [Map] to add the marshalled data
      */
-    private fun marshalIntentExtras(intent: Intent, marshalledData: MutableMap<String, Any>) {
+    @VisibleForTesting
+    internal fun marshalIntentExtras(intent: Intent, marshalledData: MutableMap<String, Any>) {
         val extraBundle = intent.extras ?: return
-        extraBundle.keySet()?.forEach { key ->
-            val newKey = when (key) {
-                LEGACY_PUSH_MESSAGE_ID -> PUSH_MESSAGE_ID_KEY
-                NOTIFICATION_IDENTIFIER_KEY -> LOCAL_NOTIFICATION_ID_KEY
-                else -> key
-            }
-            try {
-                val value = extraBundle[key]
-                if (value?.toString()?.isNotEmpty() == true) {
-                    marshalledData[newKey] = value
+
+        // First, handle known keys directly to avoid potential exceptions from keySet()
+        handleKnownKey(extraBundle, LEGACY_PUSH_MESSAGE_ID, PUSH_MESSAGE_ID_KEY, marshalledData)
+        handleKnownKey(extraBundle, NOTIFICATION_IDENTIFIER_KEY, LOCAL_NOTIFICATION_ID_KEY, marshalledData)
+
+        // Remove known keys from bundle after processing
+        try {
+            extraBundle.remove(LEGACY_PUSH_MESSAGE_ID)
+            extraBundle.remove(NOTIFICATION_IDENTIFIER_KEY)
```

Comment:
Code readability improvements: 
- Would be nice to hold these keys in a known keys map
- Could we read and remove each key as part of handleKnownKey
- Please indicate in the api docs what parameters are mutated by handleKnownKey

Reply:
+1 adding the keys to an immutable map for lookup &  removing the known key from the bundle once handled. Looks like we always remove it irrespective of whether the handling is successful or not.

Reply:
Yes, it's intended to remove that data as they're only used by our SDK.